test/master-playlist-controller.test.js

```diff
@@ -1067,6 +1067,190 @@ function(assert) {
                'less than low water line');
 });
 
+QUnit.test('does not get stuck in a loop due to inconsistent network/caching',
+function(assert) {
+  const mediaContents =
+    '#EXTM3U\n' +
+    '#EXTINF:10\n' +
+    '0.ts\n' +
+    '#EXTINF:10\n' +
+    '1.ts\n' +
+    '#EXTINF:10\n' +
+    '2.ts\n' +
+    '#EXTINF:10\n' +
+    '3.ts\n' +
+    '#EXT-X-ENDLIST\n';
+
+  const segmentLoader = this.masterPlaylistController.mainSegmentLoader_;
+
+  // start on lowest bandwidth
+  segmentLoader.bandwidth = 0;
+
+  this.player.tech_.paused = () => false;
+
+  this.masterPlaylistController.mediaSource.trigger('sourceopen');
+  // master
+  this.requests.shift().respond(200, null,
+                                '#EXTM3U\n' +
+                                '#EXT-X-STREAM-INF:BANDWIDTH=10\n' +
+                                'media.m3u8\n' +
+                                '#EXT-X-STREAM-INF:BANDWIDTH=100\n' +
+                                'media1.m3u8\n');
+  // media.m3u8
+  this.requests.shift().respond(200, null, mediaContents);
+
+  let playlistLoader = this.masterPlaylistController.masterPlaylistLoader_;
+  let origMedia = playlistLoader.media.bind(playlistLoader);
+  let mediaChanges = [];
+
+  this.masterPlaylistController.masterPlaylistLoader_.media = (media) => {
+    if (media) {
+      mediaChanges.push(media);
+    }
+    return origMedia(media);
+  };
+
+  let bandwidthupdateEvents = 0;
+
+  this.player.tech_.on('bandwidthupdate', () => bandwidthupdateEvents++);
+
+  this.clock.tick(1);
+
+  let segmentRequest = this.requests[0];
+
+  assert.equal(segmentRequest.uri.substring(segmentRequest.uri.length - 4),
+               '0.ts',
+               'requested first segment');
+
+  // 100ms for the segment response
+  this.clock.tick(100);
+
+  // 10 bytes in 100ms = 800 bits/s
+  this.requests[0].response = new Uint8Array(10).buffer;
+  this.requests.shift().respond(200, null, '');
```

Comment:
I think it might be a good idea to assert `SegmentLoader`s bandwidth after the segment request responses to make sure it sets to 800bps and didn't enter the cache path